src/algorithms/highlevel/crosssimilaritymatrix.cpp

```diff
@@ -0,0 +1,438 @@
+        // here we binarise and transpose the similarityY array same time in order to avoid redundant looping
+        if (similarityY[u][v] < 0) {
+          tSimilarityY[v][u] = 0.;
+        }
+        else if (similarityY[u][v] >= 0) {
+          tSimilarityY[v][u] = 1.;
+        }
+      }
+    }
+    */
+
+    for (size_t u=0; u<pdistances.size(); u++) {
+      for (size_t v=0; v<pdistances[u].size(); v++) {
+        similarityX[u][v] = percentile(pdistances[u], _kappa*100) - pdistances[u][v];
+        // std::cout << "Percentile: " << percentile(tpDistances[u], _kappa*100) << ", distValue: " << tpDistances[u][v] << ", SimValue: " << similarityY[u][v] << std::endl;
+        // here we binarise and transpose the similarityY array same time in order to avoid redundant looping
+        if (similarityX[u][v] < 0) {
+          similarityX[u][v] = 0.;
+        }
+        else if (similarityX[u][v] >= 0) {
+          similarityX[u][v] = 1.;
+        }
+      }
+    }
+
+    TNT::Array2D<Real> simX = vecvecToArray2D(similarityX);
+    TNT::Array2D<Real> simY = vecvecToArray2D(tSimilarityY);
+    TNT::Array2D<Real> csmOut = TNT::operator*(simX, simY);
+    csmOutput[0] = csmOut;
+    releaseData();
+  }
+  // should not arrive here
+  else EXEC_DEBUG("CrossSimilarityMatrix: wrong type parameter 'otiBinary'. Expect boolean type");
+  
+  return OK;
+}
+
+
+/*
+// construct time delayed embedding from streaming input
+std::vector<std::vector<Real> > CrossSimilarityMatrix::streamingFrames2TimeEmbedding(std::vector<std::vector<Real> > inputFrames, int m, int tau) {
+  // check if previous query frame vector is empty
+  if (_prevQueryFrame.empty()) {
+   return toTimeEmbedding(inputFrames, _embedDimension, _tau);
+  }
+  
+  std::vector<std::vector<Real> > currentFeature;
+  // preallocate memory
+  currentFeature.reserve(_prevQueryFrame.size() + inputFrames.size());
+  // concat previous query stream frames with the current query stream frames
+  currentFeature.insert(currentFeature.end(), _prevQueryFrame.begin(), _prevQueryFrame.end());
+  currentFeature.insert(currentFeature.end(), inputFrames.begin(), inputFrames.end());
+
+  std::vector<std::vector<Real> > outputVec;
+  outputVec.reserve(_prevQueryFrame.size());
+
+  for (size_t idx=0; idx<_prevQueryFrame.size(); idx+=tau) {
+    outputVec.push_back(toTimeEmbedding(currentFeature, m, tau)[0]); // TODO: adapt code for tau!=1 conditions
+    currentFeature.erase(currentFeature.begin() + idx);
+  }
+  if (currentFeature.empty()) EXEC_DEBUG("Couldn't find minimum amount of streams in the input");
+
+  _prevQueryFrame.clear();
+  _prevQueryFrame = currentFeature;
+  return outputVec;
+}
+*/
+
+
+} // namespace streaming
+} // namespace essentia
+
+
+// computes global averaged chroma hpcp as described in [1]
+std::vector<Real> globalAverageChroma(std::vector<std::vector<Real> >& inputFeature) {
+
+  size_t numbins = inputFeature[0].size();
```

Comment:
added a ```sumFrames``` function in ```essentiamath.h``` which returns the sum of frames over a specified axis

Reply:
Not sure about the option to specify the axis. This may be misleading because: ```sumFrames``` mean sum values across frames, not across features (just like other *Frames methods do). Do you actually use this option?

Reply:
Okay, I just tried to make it similar to numpy funcs are. Anyway, I thought it might be useful for some use-cases. 
Yes, I only need to sum the frames. So, just removed the parameter to specify the axis as you suggested.